domains/environment_domains/dns.tf

```diff
@@ -29,7 +29,7 @@ resource "azurerm_dns_txt_record" "apex" {
 
 resource "azurerm_dns_cname_record" "main" {
   depends_on = [azurerm_cdn_frontdoor_route.main]
-  for_each   = { for k in toset(var.domains) : k => k if k != "apex" }
+  for_each   = { for k in toset(var.domains) : k => k if !contains(setunion(["apex"], var.exclude_domains_cname), k) }
```

Comment:
They look like lists instead of sets. Can we use [concat](https://developer.hashicorp.com/terraform/language/functions/concat)?

Reply:
We could do, but since `toset(var.domains)` and `var.domains` is the same type as `var.excluded_domains_cname` and the output of either `setunion` would be the same as using `concat` we should be consistent.